### ISSUE TITLE
Halt does not terminate `jq`

### ISSUE THREAD

**Issue by smac89** [jqlang/jq#2368]
**Describe the bug**
From the `v1.6` jq [docs](https://stedolan.github.io/jq/manual/v1.6/#halt), it says that `halt` does the following:

> Stops the jq program with no further outputs. jq will exit with exit status 0

However, from my testing of both `halt` and `halt_error`, neither of them causes jq to terminate with no further outputs.

**To Reproduce**
Input:
```json
{"value":{"error":"no such element","message":"no such element: xxx"}}
{"value":null}
```
Command:
```sh
jq '.value | if .error then "\(.message)\n" | halt_error(1) else "" end'
```
Output:
```
no such element: xxx
""
```

**Expected behavior**
There should only be one output, seeing as after reading the first object, it causes `halt_error` to run

**Environment (please complete the following information):**
 ```
Distributor ID: Ubuntu
Description:    Ubuntu 21.10
Release:        21.10
Codename:       impish
```
 - jq version 1.6

**Additional context**

**Comment by smac89** [comment]
Just as I posted this, I found a solution [here](https://github.com/stedolan/jq/issues/1188).

**Comment by pkoppstein** [comment]
@smac89 - The issue you raise could be regarded entirely as a documentation issue. Indeed, the jq developers have generally been quite definite about not wanting to break the stream-processing character of jq - i.e., assuming that input has not been "slurped", and that each input is well-formed, each input is processed separately.

**Comment by itchyny** [comment]
This has been resolved by #2667.
